File: RiotGames/Dto/Summoner/SummonerDto.cs
using Newtonsoft.Json;

namespace potg.RiotGames.Dto.Summoner;

public record SummonerDto
{
    [JsonProperty("accountId")]
    public string AccountId { get; set; }

    [JsonProperty("profileIconId")]
    public int ProfileIconId { get; set; }

    [JsonProperty("revisionDate")]
    public ulong RevisionDate { get; set; }

    [JsonProperty("name")]
    public string Name { get; set; }

    [JsonProperty("id")]
    public string Id { get; set; }

    [JsonProperty("puuid")]
    public string Puuid { get; set; }

    [JsonProperty("summonerLevel")]
    public int SummonerLevel { get; set; }
}

File: src/models/dto/riot/lol/summoner/summoner.dto.ts
import { ApiProperty } from "@nestjs/swagger";
import { MaxLength, IsString, IsNumber, Length } from "class-validator";

export class SummonerDto {
  @MaxLength(56)
  @IsString()
  @ApiProperty()
  accountId: string;

  @IsNumber()
  @ApiProperty()
  profileIconId: number;

  @IsNumber()
  @ApiProperty()
  revisionDate: number;

  @IsString()
  @ApiProperty()
  name: string;

  @MaxLength(63)
  @IsString()
  @ApiProperty()
  id: string;

  @Length(78)
  @IsString()
  @ApiProperty()
  puuid: string;

  @IsNumber()
  @ApiProperty()
  summonerLevel: number;
}
